### PR TITLE
Jorge freddy rai

### PR DESCRIPTION
**Pull request by alexRai98**

### CODE REVIEW

File: blog.html [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    <meta http-equiv="X-UA-Compatible" content="ie=edge" />
+    <title>Codeable Blog</title>
+    <link rel="stylesheet" href="/css/normalize.css" />
+    <link rel="stylesheet" href="/css/main.css" />
+    <link rel="stylesheet" href="/css/blog.css" />
+  </head>
+  <body>
+    <header>
+      <div class="wrapper">
+        <b>Codeable News</b>
```
Comment (julionav) [inline blog.html]: The name of the site is probably the most important heading of it.
What do you think about using a heading here instead?
More on headings: https://developer.mozilla.org/es/docs/Web/HTML/Elemento/Elementos_t%C3%ADtulos

Comment (julionav) [reply blog.html]: Another note here is that it's preferred to use `strong` over `b`.
[`strong` has a semantic meaning](https://developer.mozilla.org/es/docs/Web/HTML/Elemento/strong), it gives emphasis to a sentence.
`b` just mean "bold", which is a visual aspect without any semantic meaning

File: blog.html [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        <nav>
```
Comment (julionav) [inline blog.html]: Nice `nav` use 💯

File: blog.html [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <meta http-equiv="X-UA-Compatible" content="ie=edge" />
+    <title>Codeable Blog</title>
+    <link rel="stylesheet" href="/css/normalize.css" />
+    <link rel="stylesheet" href="/css/main.css" />
+    <link rel="stylesheet" href="/css/blog.css" />
+  </head>
+  <body>
+    <header>
```
Comment (julionav) [inline blog.html]: Nice work using the `header` element here

File: blog.html [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        <nav>
+          <ul>
+            <li><a href="/">Home</a></li>
+            <li><a href="/#pricing">Pricing</a></li>
+            <li><a href="/">Contact</a></li>
+          </ul>
+        </nav>
+      </div>
+    </header>
+    <main class="container">
```
Comment (julionav) [inline blog.html]: Good use of the generic `container`.
Remember to always reserve generic names like this for styles that are highly reusable. In this case, if the you have the same width container in other pages, you can use a name like this.

File: blog.html [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+      <article class="blog-post">
+        <h2>COMPUTER SCIENCE</h2>
+        <h3>How to increase the efficiency of your algorithms</h3>
+        <ul class="tags">
```
Comment (julionav) [inline blog.html]: I don't believe this can be considered to be `tags`. Looks more like the metadata of the post.

File: blog.html [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+          <li>By <b>Marvin Carrasco</b></li>
+          <li>March 30, 2020</li>
+          <li>6 comments</li>
+        </ul>
```
Comment (julionav) [inline blog.html]: A good practice is to wrap the entire header of an article in a `header` tag

File: blog.html [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+            <li><a href="/">Home</a></li>
+            <li><a href="/#pricing">Pricing</a></li>
+            <li><a href="/">Contact</a></li>
+          </ul>
+        </nav>
+      </div>
+    </header>
+    <main class="container">
+      <article class="blog-post">
+        <h2>COMPUTER SCIENCE</h2>
```
Comment (julionav) [inline blog.html]: Here, you are giving more importance to `Computer Science` with the `h2`, which is more a category for the post, than to the actual title of the post.
Beside looking like `titles` and being `big`, headings define the document outline, which is one of the primary ways screen readers see the structure of the site.
Yo can read more about `Document` outlines here: https://www.internetingishard.com/html-and-css/semantic-html/#the-document-outline

Short version are that the headings define the name of a section, so you need to ask yourself. What is the name of this section (or article) that contains my blog post?

Comment (julionav) [reply blog.html]: Another way to look at this is to use a document outliner tool like https://gsnedders.html5.org/outliner/
and paste your entire HTML. You will see the outline of it and you can make a few changes so It looks like something that makes sense.
Currently it looks like this:

Outline

P.D: El "Document Outline" es basicamente el "indice" del documento. Imaginate que tu sitio es un archivo de word

File: blog.html [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+        <h3>How to increase the efficiency of your algorithms</h3>
+        <ul class="tags">
+          <li>By <b>Marvin Carrasco</b></li>
+          <li>March 30, 2020</li>
+          <li>6 comments</li>
+        </ul>
+        <img src="/assets/blog-algorithms.png" alt="" />
```
Comment (julionav) [inline blog.html]: Always add the `alt` property to an img. It's good for SEO and it makes the site more accesible.

File: css/blog.css [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+
+header li {
+    display: inline-block;
+    padding: 0 32px;
+}
+
+header a {
+    text-decoration: none;
+    color: inherit;
+}
```
Comment (julionav) [inline css/blog.css]: Setting styles to root components is, in general, not recommended. Unless the styles are 100% sure to be always the ones you want everywhere. This is mostly because element selectors have a very strong specificity and it is a challenge to override those styles later.

Even in those cases, it is better to use classes. If you want to reuse the styles you can just reuse the class.

Normally, setting styles to root elements is reserved to [CSS resets](https://css-tricks.com/reboot-resets-reasoning/)

P.D: I know Diego asked to not add many classes and it is ok during this project. I will always ty do my CR the most similar to how I would do it in production (real) project.

File: css/blog.css [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+header {
+    background: #929ECC;
+    color: #fff;
+    position: fixed;
+    left: 0;
+    right: 0;
+    top: 0;
+}
+
+header .wrapper {
```
Comment (julionav) [inline css/blog.css]: Here, you could have added the styles directly to `.wrapper`.
Adding more specific selectors make the styles harder to override later (and you will probably needed in a real project)

File: css/blog.css [hunk 11]
```diff
@@ -0,0 +1,106 @@
...
+    display: inline-block;
+    font-size: 14px;
+    line-height: 16px;
+}
+
+.blog-post .tags li::after {
+    content: '\25C6';
+    font-size: 8px;
+    padding: 0 12px;
+}
```
Comment (julionav) [inline css/blog.css]: I personally prefer (and it is a style to write CSS) to avoid composed rules as much as possible and just add more classes, instead of adding selectors that involves parents.

Adding composed selectors make (in my experience) CSS harder to maintain because you create dependencies between classes and elements. In this case, if you replace the `b` from `.blog-post` for an `strong` (as I recommended above), you would need to also come here to your HTML and update the selector.

This also creates a dependency between the structure of your HTML and the CSS that you write. System that depend on each other are harder to maintain because a change that you do on one, can also affect the other. Now imagine that you have many CSS files with many HTML files that all depend in each other; it is the perfect formula for a disaster and headaches.

File: css/blog.css [hunk 12]
```diff
@@ -0,0 +1,106 @@
...
+
+.blog-post p {
+    line-height: 1.5;
+    margin-top: 32px;
+    margin-bottom: 24px;
+    max-width: 700px;
+}
+
+.blog-post p+a {
```
Comment (julionav) [inline css/blog.css]: Same as above.
In general, avoid complex rules if you can! the `+` selector isn't very complex but even that takes me a bit to remember what it does (sometimes I even need to google it!)

In this days where developers need to know about so make key parts of the ecosystem, it's hard to remember the little things.

File: index.html [hunk 13]
```diff
@@ -4,9 +4,57 @@
...
+          <ul>
+            <li>Just one enrollment payment</li>
+            <li>Pay when you get a job</li>
+            <li>Your salary should be greater than $750</li>
+          </ul>
+
+          <button class="btn-pricing">Get Started</button>
+        </div>
+      </div>
+    </section>
```
Comment (julionav) [inline index.html]: This HTML structure creates a really nice Documment outline. Nice work!

File: css/blog.css (hunk 9 above)
Comment (Deprofets) [reply css/blog.css]: I agree, styling root components should be avoided unless there are base styles for them, but in this case body padding is required on the blog CSS to compensate for the fixed positioning of the header.

File: css/blog.css (hunk 10 above)
Comment (Deprofets) [reply css/blog.css]: True but if I did would you've been able to tell where the wrapper was from reading the CSS alone?

File: css/blog.css (hunk 11 above)
Comment (Deprofets) [reply css/blog.css]: I started using this approach recently because it reduces the final HTML size but now that I think about it spamming classes everywhere should make both files independent and easier to maintain for a team.

File: css/blog.css (hunk 12 above)
Comment (Deprofets) [reply css/blog.css]: `p+a` targets every link after a paragraph, not 100% sure about this one but you might be right. Anyway just changing the whole thing to `.blog-post--link` should make this more readable for everyone.

File: blog.html [hunk 14]
```diff
@@ -0,0 +1,64 @@
...
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <meta http-equiv="X-UA-Compatible" content="ie=edge" />
+    <title>Codeable Blog</title>
+    <link rel="stylesheet" href="/css/normalize.css" />
+    <link rel="stylesheet" href="/css/main.css" />
+    <link rel="stylesheet" href="/css/blog.css" />
+  </head>
+  <body>
+    <header>
+      <div class="wrapper">
```
[the remark this answers is not in the document]
Comment (Deprofets) [reply blog.html]: I forgot to change this to `header-wrapper`.

File: blog.html (hunk 1 above)
Comment (Deprofets) [reply blog.html]: Yes I should've used a h1 instead because the site **is** about Codeable news, my initial reasoning for this was that I thought this was something like a logo.

File: css/blog.css (hunk 9 above)
Comment (julionav) [reply css/blog.css]: You can just add a class to the container of the blog posts to add that padding

Comment (Deprofets) [reply css/blog.css]: Yes but then I wouldn't be able to reuse that class elsewhere

File: blog.html [hunk 15]
```diff
@@ -0,0 +1,68 @@
...
+    <main class="container">
+      <article class="blog-post">
+        <header>
+          <span class="blog-post-category">COMPUTER SCIENCE</span>
+          <h1 class="blog-post-title">How to increase the efficiency of your algorithms</h1>
+          <ul class="blog-post-metadata">
+            <li>By <b>Marvin Carrasco</b></li>
+            <li>March 30, 2020</li>
+            <li>6 comments</li>
+          </ul>
```
Comment (diegotc86) [inline blog.html]: Using `<ul>` tags here is not so common practice. An `<span>` could be enough. You can use an UNICODE symbol to display those black diamonds between each phrase. You cab also use the `<time>` tag for the date. Something like this (maybe with some spans are not necessary):
```html
<span class="blog-post-metadata">
 <span>By <strong>Marvin Carrasco</strong>/span>
 <span>&#9670;</span>
 <time datetime="2020-03-30">March 30, 2020</time>
 <span>&#9670;</span>
 <span>6 comments</span>
</span>
```

File: blog.html [hunk 16]
```diff
@@ -0,0 +1,68 @@
...
+        </header>
+        <img src="/assets/blog-algorithms.png" alt="Algos" />
+        <p class="blog-post-summary">
+          In computer science, algorithmic efficiency is a property of an
+          algorithm which relates to the number of computational resources used
+          by the algorithm. An algorithm must be analyzed to determine its
+          resource usage, and the efficiency of an algorithm can be measured
+          based ...
+        </p>
```
Comment (diegotc86) [inline blog.html]: There is an special HTML entity you can use instead of `...` at he end of the paragraphs. Is the ellipsis: `&hellip;`

File: css/blog.css [hunk 17]
```diff
@@ -0,0 +1,113 @@
...
+}
+
+.site-header-wrapper {
+    max-width: 1170px;
+    margin: 0 auto;
+    padding: 16px 32px;
+}
+
+.site-title{
+    font-size: 16px;
```
Comment (diegotc86) [inline css/blog.css]: It looks good but not exactly match the design. The font-size required was 20px.

File: css/blog.css [hunk 18]
```diff
@@ -0,0 +1,113 @@
...
+.site-header ul {
+    list-style-type: none;
+    margin: 0 -32px;
+    padding: 0;
+}
+
+.site-header li {
+    display: inline-block;
+    padding: 0 32px;
+}
```
Comment (diegotc86) [inline css/blog.css]: The font-size should be specified here or in the `.site-header a` rule

File: css/blog.css [hunk 19]
```diff
@@ -0,0 +1,113 @@
...
+
+.blog-post-link {
+    display: inline-block;
+    text-decoration: none;
+    font-weight: 500;
+    color: inherit;
+    border: 1px solid #e5e5e5;
+    line-height: 0.94;
+    padding: 18px 40px;
+}
```
Comment (diegotc86) [inline css/blog.css]: Hi! Here you managed to get the right width of the button using the padding property. There is no doubt that the button should have a padding, we don't want the text to touch the borders but, what would happen if the text is a little bigger? Well, your button will be bigger automatically and that's not great:
image
It would be better to work with a fixed width her :)

File: css/blog.css [hunk 20]
```diff
@@ -0,0 +1,113 @@
...
+.blog-post-metadata li:last-child::after {
+    content: '';
+}
+
+.blog-post-summary {
+    line-height: 1.5;
+    margin-top: 32px;
+    margin-bottom: 24px;
+    max-width: 700px;
+}
```
Comment (diegotc86) [inline css/blog.css]: max-width give the site some type of responsive behavior but is not needed really (because not all the other elements have responsive techniques in place). The paragraph shrinks while reducing the browser size but the image for example stays the same size. width would be enough 👍

File: index.html [hunk 21]
```diff
@@ -4,9 +4,57 @@
...
+        <a class="btn-pricing" href="#pricing">Pricing</a>
+        <a class="btn-blog" href="./blog.html">Visit our blog</a>
+      </div>
+      <div class="container-image">
+        <img src="./assets/landing-img1.png" alt="Landing" />
+      </div>
+    </section>
+    <!-------------Start Pricing ---------->
+
+    <section id="pricing">
```
Comment (diegotc86) [inline index.html]: Good use of `id` to link this section with the button `pricing`

File: index.html [hunk 22]
```diff
@@ -4,9 +4,57 @@
...
+          <ul>
+            <li>Study on huge discount.</li>
+            <li>Get a job only if you like</li>
+            <li>Enjoy the same benefits</li>
+          </ul>
+
+          <button class="btn-pricing">Get Started</button>
+        </div>
+        <div class="plans">
+          <h2>ISA <br>$9,000</h2>
```
Comment (diegotc86) [inline index.html]: Nice use of `<br>` here. The other way was using the width property to force the text break to another line like this
```html
<h2 style="
    width: 250px;
    margin: 36px auto;
">Up front $6,000</h2>
```

File: css/main.css [hunk 23]
```diff
@@ -0,0 +1,128 @@
...
+  margin-top: 40px;
+  text-align: left;
+  margin-bottom: 90px;
+  padding-left: 100px;
+}
+.plans li {
+  font-size: 20px;
+  margin-right: 60px;
+  margin-bottom: 40px;
+}
```
Comment (diegotc86) [inline css/main.css]: margin-bottom here seems to much, 30px at max according to design?

File: css/main.css [hunk 24]
```diff
@@ -0,0 +1,128 @@
+@import url("https://fonts.googleapis.com/css?family=Overpass|Roboto&display=swap");
+.landing > * {
+  box-sizing: border-box;
+  margin: 0;
+  padding: 0;
+}
+html,
+body {
+  font-family: "Roboto", sans-serif;
+}
```
Comment (diegotc86) [inline css/main.css]: Maybe add here a max-width to respect the design. Maybe the body is not the best place but without a main wrapper there is no other space.
```html
max-width: 1440px;
margin: auto;
```

File: css/blog.css (hunk 9 above)
Comment (julionav) [reply css/blog.css]: It is ok to just use a class once 😄

File: css/blog.css (hunk 10 above)
Comment (julionav) [reply css/blog.css]: Yes! if the name of the class were a little more specific like `.navbar-wrapper`

File: css/blog.css (hunk 11 above)
Comment (julionav) [reply css/blog.css]: Having a big HTML size is ok. 

In the future you will work with template engines or component libraries, so you don't need to have the entire HTML in a single file.
